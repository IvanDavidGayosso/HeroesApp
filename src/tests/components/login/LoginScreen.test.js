import { mount } from "enzyme"
import { AuthContext } from "../../../auth/AuthContext"
import { LoginScreen } from "../../../components/login/LoginScreen"
import { types } from "../../../types/types"


describe('Pruebas eb <LoginScreen />', () => {

    const history = {
        replace: jest.fn()
    }

    const valueContex = {
        dispatch: jest.fn()
    }

    const wrapper = mount(
            <AuthContext.Provider value={ valueContex }>
                <LoginScreen history={ history } />
            </AuthContext.Provider >
            
        );
    
    test('debe de mostrarse correctamente ', () => {
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.container').exists() ).toBe(true);

    })

    test('debe de realizar el dispatch y la navegaciÃ³n ', () => {
        
        wrapper.find('button').prop('onClick')();

        expect( history.replace ).toHaveBeenCalled();
        expect( valueContex.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'Ivan David'
            }
        });

    })
    
    

})
