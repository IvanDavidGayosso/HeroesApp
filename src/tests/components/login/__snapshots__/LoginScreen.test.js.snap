// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas eb <LoginScreen /> debe de mostrarse correctamente  1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
